src/FlowerShop.Core/Entities/Order.cs

```diff
@@ -9,18 +9,54 @@ public class Order
         public string PhoneNumber { get; private set; }
         public decimal TotalPrice { get; private set; }
         public string Comment { get; private set; }
+        public string Email { get; private set; }
         public OrderStatus OrderStatus { get; private set; }
-        public int AddressId { get; private set; }
+        public int? AddressId { get; private set; }
         public Address Address { get; private set; }
-        public int UserId { get; private set; }
+        public int? UserId { get; private set; }
         public User User { get; private set; }
 
+        // Creation fields
+        public string FirstName { get; set; }
```

Comment:
Why ? Order already has user which has the fields. Also it has adress which also uses the fieds. This is an entity that will be mapepd to database and it will create uneccesarry fields, also i'm pretty sure this goes Against SOLID principles.

Reply:
But what about cases when the user is NOT registered? We can't really pin an Address Entity to it.

Furthermore, if a user deleted an address - a completed order will have no address fields.

Reply:
User should always be registered when purchasing something to because we need to know who ordered the order. We can assign the address after we know the user. For the second part that's why there's a thing called Soft delete.

Reply:
But there's a requirement/user-story for being able to place orders WITHOUT a registered account (from the original task text)

Reply:
I was thinking an email + phoneNumber would suffice for identifying WHO bought what (when dealing with unregistered users)

Reply:
The we need to change how Address behaves so we can assign it without an user :/ . Usually as far as i've seen apps which handle Anonymous users still create a user object with an Anonymous user object.

Reply:
We're gonna be making "shadow-accounts" now? Kinda stepping into Facebook area :D 

Is it efficient to try and make anonymous users now? Or should we stick to ordering by email/phone number for unregistered users?